Sources/Web3Core/KeystoreManager/BIP32Keystore.swift

```diff
@@ -166,54 +169,75 @@ public class BIP32Keystore: AbstractKeystore {
     }
 
     public func createNewCustomChildAccount(password: String, path: String) throws {
-        guard let decryptedRootNode = try getPrefixNodeData(password) else {
+        guard let decryptedRootNode = try getPrefixNodeData(password),
+              let keystoreParams = keystoreParams else {
             throw AbstractKeystoreError.encryptionError("Failed to decrypt a keystore")
         }
         guard let rootNode = HDNode(decryptedRootNode) else {
             throw AbstractKeystoreError.encryptionError("Failed to deserialize a root node")
         }
+
         let prefixPath = self.rootPrefix
         var pathAppendix: String?
+
         if path.hasPrefix(prefixPath) {
-            let upperIndex = (path.range(of: prefixPath)?.upperBound)!
-            if upperIndex < path.endIndex {
-                pathAppendix = String(path[path.index(after: upperIndex)])
+            if let upperIndex = (path.range(of: prefixPath)?.upperBound), upperIndex < path.endIndex {
+                pathAppendix = String(path[path.index(after: upperIndex)..<path.endIndex])
             } else {
                 throw AbstractKeystoreError.encryptionError("out of bounds")
             }
 
-            guard pathAppendix != nil else {
+            guard var pathAppendix = pathAppendix else {
```

Comment:
Well, this is wrong.
Below you will edit only a local version of `pathAppendix`. Local and visible only inside of `if path.hasPrefix(prefixPath) { ... }`.

When you leave that scope starting with line 199 you will reference absolutely different `pathAppendix`. Change it back to `pathAppendix != nil`.